core/domain/collection_domain.py

```diff
@@ -1074,6 +1086,24 @@ def add_skill(self, skill_name):
         skill_id = CollectionSkill.get_skill_id_from_index(self.next_skill_id)
         self.skills[skill_id] = CollectionSkill(skill_id, skill_name, [])
         self.next_skill_id += 1
+        return skill_id
+
+    def get_skill_id_from_skill_name(self, skill_name):
+        """Gets the skill id from the skill name.
+
+        Args:
+            skill_name: str. The name of the skill.
+
+        Returns:
+            str. The id of the skill.
+        """
+        index = 0
+        for _, skill in self.skills.iteritems():
```

Comment:
This doesn't seem right to me? Dicts are unordered.

Isn't self.skills mapping skill_id to skill? Can't you get the skill_id based on that? I don't understand why we're introducing a "skill index".

(As an aside, this is why these methods do need tests...)

Reply:
Updated.